### ISSUE TITLE
Travis CI: reorder to speed up detection of first error

### ISSUE THREAD

**Issue by vlcinsky** [nameko/nameko#630]
Currently, travis runs large set of tests and it takes quite some time.

It would be great, if time to detection of first error could be shorter.

Currently it e.g. builds documentation which could be safely postponed as i assume, most development happens with code and not with the docs.

My proposal for order of initial test environments is:

```yaml
matrix:
  include:
# start with most popular predictable env. (py36)
## first detect functional issues and coverage problems
    - stage: test
      python: 3.6
      env: TOX_ENV=py36-pinned-lib
## then do static checks
    - stage: static
      python: 3.6
      env: TOX_ENV=py36-static
# repeat for py27
## first detect functional issues and coverage problems
    - stage: test
      python: 2.7
      env: TOX_ENV=py27-pinned-lib
## then do static checks
    - stage: static
      python: 2.7
      env: TOX_ENV=py27-static
# Build the docs
    - stage: docs
      python: 2.7
      env: TOX_ENV=docs
# Continue with whatever seems relevant

```

**Comment by mattbennett** [comment]
This makes sense to me, except I would do static analysis before tests within each env. Static analysis is fast and fails often, so putting it first would decrease time to first error detection.

**Comment by vlcinsky** [comment]
I was also hesitating, if static analysis first or not, I am fine with running it first.

Another improvement could be to remove python 3.4 testing for 3.0.0rc. No speed up (apart from total duration), but why not saving resources for other tests.

Should I create a PR?

**Comment by mattbennett** [comment]
Yes please!

**Comment by vlcinsky** [comment]
My plan is to:
- test in order py36, 27, 35 for `static` and `latest-lib` (latest seems more important than pinned)
- then `doc`
- then the rest (following the order of python versions)
- then the rest
- remove py34 (as we commit the change into `v3.0.0-rc`

**Comment by mattbennett** [comment]
Fixed in #631
